### ISSUE TITLE
Enable users to save a search and view/open/delete their saved searches

### ISSUE THREAD

**Issue by mxk0** [HHS/simpler-grants-gov#3119]
### Summary 

Today, users must visit simpler.grants.gov and enter a search to see the latest results for a particular keyword/filter combination. This results in a lot of manual repetitive effort, and may cause grant applicants to miss key opportunities if they're on vacation, forget to run their weekly search, etc. By allowing users to save and subscribe to searches, we can save them time and ensure they learn about relevant opportunities as soon as they're published.

This feature already exists on Grants.gov. Usage is relatively low on Grants.gov – ~3000 active users use the feature – but we suspect current usage reflects poor UX and discoverability. The user need and a proposed solution also [came out of the Collaborative Coding and Design Challenge](https://docs.google.com/presentation/d/1BAhUlzwG6ET7DSWJHNL3JLGMC5b9j29K-GqIv_1Hr_U/edit#slide=id.p).

### Acceptance criteria

- [ ] Users can save a set of search criteria (terms, filters, etc.)
- [ ] Users can view their saved searches
- [ ] Users can open a saved search
- [ ] Users can delete a saved search

### Dependencies

- This feature depends on #2640 

### Not in scope for first iteration

- Users can edit a saved search. For now, this use case will be covered by deleting a search and saving a new one, since it should be easy to open a search and re-save.
- Users can share a saved search. Users can share a search and the sharee can save a new search. This should cover the use case for now.
- Email notifications about new opportunities matching a saved search. This will happen in a separate deliverable later in Quad 2.
- We may explore team-level saved searches at some point, but that is not in scope for now.
- Saving the result set from a search.

**Comment by mxk0** [comment]
### Notes from 1/30 kickoff (not comprehensive, lots of [notes in Figma](https://www.figma.com/design/FGxWtAgToKhehLJCiuy1zL/Simpler.Grants.gov-Project?node-id=1809-14763&p=f&t=YLapdvMZxeLLTvgl-0) too)

- Proposed update to opportunity status filter: when do we do this? (Need a ticket, @mxk0 to follow up)
- Enable users to edit the name of a saved search? Probably not for v1
- Enable someone to edit terms/filters in a saved search? Probably not for v1
- What exactly is getting saved?
  - Keywords, filters, sorting
  - Don't save pagination. Always default back to page 1 when the user loads a saved search.
- What about for copy? Do we persist pagination for a copied search URL, or always send the user back to the first page?
  - Yes, persist pagination. Not clear what the ideal UX is, and would be hacky to strip out page.

cc @snyder-ux @doug-s-nava @chouinar @acouch

**Comment by mxk0** [comment]
### Notes from 1/30 kickoff (not comprehensive, lots of [notes in Figma](https://www.figma.com/design/FGxWtAgToKhehLJCiuy1zL/Simpler.Grants.gov-Project?node-id=1809-14763&p=f&t=52zfkGi6qrO3NRmW-0) too)

- For the quick filter dropdown, do we load the saved search when the user chooses something from the dropdown, or do they select something and then click "Apply"?
  - Maybe accessibility concerns of loading something when the user clicks an option, better to have the user click "Apply"
  - What does "Copy Search" copy if the user selected a different quick filters option but hasn't yet clicked "Apply"?

**Comment by andycochran** [comment]
We discussed not having an apply button since our filters do not work in that way. It could result in the opposite problem of the user not knowing whether the choice is applied or just selected. It'd also make the copy button confusing during that intermediate state. 

I wonder if this shouldn't be a `<select>` element. But instead open a menu/pane/dropdown that shows more detail about the list of filters you've saved. An apply button next to each option might make more sense.

**Comment by snyder-ux** [comment]
@doug-s-nava drafted this [Saved Search Engineering Tickets](https://docs.google.com/document/d/1fTai3r0CI71NJUfKZGiqjWWNxoTEYohavWeGzKvVnYc/edit?tab=t.0) after walking through designs, to help with organizing and creating implementation tickets.
Once change since then: Navigation updated so saved opportunities and saved search queries are on _different_ pages.
